framework/build.sh

```diff
@@ -7,25 +7,27 @@ cd build
 
 cmake \
   -G Ninja \
-  -DUSE_SYSTEM_EIGEN=ON \
-  -DUSE_JEMALLOC=OFF \
-  -DENABLE_OPENGL=OFF \
-  -DENABLE_DOCS=OFF \
-  -DENABLE_MANTIDPLOT=FALSE \
-  -DENABLE_WORKBENCH=FALSE \
-  -DENABLE_OPENCASCADE=FALSE \
+  -DCMAKE_BUILD_TYPE=Release \
+  -DUSE_JEMALLOC=ON \
+  -DUSE_CCACHE=OFF \
   -DPython_EXECUTABLE="$CONDA_PREFIX/bin/python" \
   -DCMAKE_PREFIX_PATH=$CONDA_PREFIX \
   -DCMAKE_INSTALL_PREFIX=$CONDA_PREFIX \
-  -DCMAKE_SKIP_INSTALL_RPATH=ON \
+  -DCMAKE_OSX_DEPLOYMENT_TARGET=$OSX_VERSION \
+  -DHDF5_ROOT=$CONDA_PREFIX \
+  -DCMAKE_OSX_SYSROOT="/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX$OSX_VERSION.sdk" \
```

Comment:
How does `OSX_VERSION` end up getting defined?

Reply:
It needs to be set as an environment variable. That will need to be present for a subsequent jenkins configuration to work. Though this should also be documented in the relevant readme of this repository.

Reply:
@peterfpeterson  one of the main changes is having `-DCMAKE_SKIP_INSTALL_RPATH` defaulted to OFF - we need this for the OSX builds but neither @martyngigg or myself understand why it was needed in the first place (or even how it worked without it).

Reply:
I have very vague memory of scrubbing cmake for all of the RPATH stuff when we originally got paraview bundled in. I don't entirely recall what the problem was though. OSX used to be very persistent about leaving the RPATH including system locations. I recall a ruby script running during the final packaging stage, but fail to find it.

Reply:
It would be great if someone could do some additional testing that removing this flag does not break things on Linux. Infact with the `test` changes I just added, just running `conda build` would now fail if the package was broken on linux.